Sources/NIOSSL/SSLContext.swift

```diff
@@ -502,6 +509,89 @@ extension NIOSSLContext {
     }
 }
 
+// For accessing STACK_OF(SSL_CIPHER) from a SSLContext
+extension NIOSSLContext {
+    /// A collection of buffers representing a STACK_OF(SSL_CIPHER)
+    struct NIOTLSCipherBuffers {
+        private let basePointer: OpaquePointer
+
+        fileprivate init(basePointer: OpaquePointer) {
+            self.basePointer = basePointer
+        }
+    }
+
+    /// Invokes a block with a collection of pointers to STACK_OF(SSL_CIPHER).
+    ///
+    /// The pointers are only guaranteed to be valid for the duration of this call.  This method aligns with the RandomAccessCollection protocol
+    /// to access UInt16 pointers at a specific index.  This pointer is used to safely access id values of the cipher to create a new NIOTLSCipher.
+    func withStackOfCipherSuiteBuffers<Result>(_ body: (NIOTLSCipherBuffers?) throws -> Result) rethrows -> Result {
+        guard let stackPointer = CNIOBoringSSL_SSL_CTX_get_ciphers(self.sslContext) else {
+            return try body(nil)
+        }
+        return try body(NIOTLSCipherBuffers(basePointer: stackPointer))
+    }
+
+    /// Cipher suites derived from the default ciphers on the SSLContext matched against the available ciphers in NIOTLSCipher.
```

Comment:
This comment isn't strictly accurate: we will represent them all, we just won't necessarily have nice names for some of them.

Reply:
Resolved in https://github.com/apple/swift-nio-ssl/pull/293/commits/dc9625669fc700a8f1cb8a064ad4c59a84fc6563